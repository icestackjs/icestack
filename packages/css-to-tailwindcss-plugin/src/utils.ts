import path from 'node:path'
import fs from 'node:fs'
export function resolvePath(p: string, cwd: string = process.cwd()) {
  return path.isAbsolute(p) ? p : path.resolve(cwd, p)
}

export function ensureDir(dir: string) {
  if (fs.existsSync(dir)) {
    const stat = fs.statSync(dir)
    if (!stat.isDirectory()) {
      fs.mkdirSync(dir)
    }
  } else {
    fs.mkdirSync(dir)
  }
}
