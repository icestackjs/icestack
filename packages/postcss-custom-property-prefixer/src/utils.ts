import { PrefixFunction, UserDefinedOptions } from './types'

export function matchCustomPropertyFromValue(str: string, cb: (arr: RegExpExecArray, index: number) => void) {
  let arr: RegExpExecArray | null
  let index = 0
  const regex = /var\(\s*(--\w[\w-]*)/g
  while ((arr = regex.exec(str)) !== null) {
    cb(arr, index)
    index++
  }
}

export function makeCustomProperty(customProperty: string, prefix: string) {
  return `--${prefix}${customProperty.slice(2)}`
}

export const postcssPlugin = 'postcss-custom-property-prefixer'

export function makePrefixFunction(prefix: UserDefinedOptions['prefix']): PrefixFunction {
  return typeof prefix === 'string'
    ? () => {
        return prefix
      }
    : prefix
}
