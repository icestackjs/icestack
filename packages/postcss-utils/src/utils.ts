import type { ChildNode } from 'postcss'
import sassParser from 'postcss-scss'
import selectorParser from 'postcss-selector-parser'
import { set, get } from 'lodash'
import type { CssValue, CssInJs } from '@icestack/types'

const defaultSelectorParser = selectorParser()

export function compressCssSelector(selectors: string) {
  return defaultSelectorParser.processSync(selectors, { lossless: false })
}

export function recursiveNodes(nodes: ChildNode[], result: Record<string, any> = {}) {
  for (const node of nodes) {
    switch (node.type) {
      case 'atrule': {
        if (node.name === 'apply') {
          const v = get(result, 'apply')
          if (Array.isArray(v)) {
            v.push(node.params)
          } else {
            set(result, 'apply', [node.params])
          }
        } else {
          const selector = `@${node.name} ${node.params}`
          result[selector] = {}
          recursiveNodes(node.nodes ?? [], result[selector])
        }

        break
      }
      case 'rule': {
        const s = compressCssSelector(node.selector)
        if (result[s] === undefined) {
          result[s] = {}
        }

        recursiveNodes(node.nodes, result[s])
        break
      }
      case 'decl': {
        set(result, `css.${node.prop}`, node.value + (node.important ? ' !important' : ''))
        break
      }
    }
  }
  return result
}

export function transformCss2Js(css: string): Record<string, any>
export function transformCss2Js(css: string | CssInJs): Record<string, any>
export function transformCss2Js<T>(css: T) {
  if (typeof css === 'string') {
    const root = sassParser.parse(css)
    const result = recursiveNodes(root.nodes)
    return result
  }
  return css
}

export function mapCss2JsArray(value?: CssValue) {
  if (value) {
    return Array.isArray(value) ? value.map((x) => transformCss2Js(x)) : [transformCss2Js(value)]
  }
  return []
}
