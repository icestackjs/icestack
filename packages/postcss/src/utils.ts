export * from '@icestack/postcss-utils'
