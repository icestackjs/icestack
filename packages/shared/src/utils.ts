import { isObject, set, get } from 'lodash'
import postcss from 'postcss'
import selectorParser from 'postcss-selector-parser'

import { createDefu } from 'defu'

const defuOverrideArray = createDefu((obj, key, value) => {
  if (Array.isArray(obj[key]) && Array.isArray(value)) {
    obj[key] = value
    return true
  }
})

const defuOverrideApplyCss = createDefu((obj, key, value) => {
  if ((key === 'css' || key === 'apply') && typeof obj[key] === 'object' && typeof value === 'object') {
    obj[key] = value
    return true
  }
})

export { defuOverrideArray, defuOverrideApplyCss }

const defaultSelectorParser = selectorParser()

export function compressCssSelector(selectors: string) {
  return defaultSelectorParser.processSync(selectors, { lossless: false })
}

export function expandTypes(types: string[], fn: (typeName: string) => { key: string; value: object }) {
  return types.reduce<Record<string, object>>((acc, cur) => {
    const { key, value } = fn(cur)
    acc[key] = value
    return acc
  }, {})
}

export function preprocessCssInJs(obj: Record<string, any>, res: Record<string, any> = {}) {
  if (typeof obj !== 'object') {
    return res
  }
  const keys = Object.keys(obj)

  for (const key of keys) {
    const value = obj[key]
    if (key === 'apply') {
      if (typeof value === 'string') {
        if (Array.isArray(res[key])) {
          res[key].push(value.split(' '))
        } else {
          res[key] = [value.split(' ')]
        }
      } else if (Array.isArray(value)) {
        // has been handled
        // preprocessCssInJs case 2

        for (const applyStr of value) {
          if (typeof applyStr === 'string') {
            if (Array.isArray(res[key])) {
              res[key].push(applyStr.split(' '))
            } else {
              res[key] = [applyStr.split(' ')]
            }
          }
        }

        // res[key] = value
      }
      // do nothing
    } else if (key === 'css') {
      res[key] = value
    } else if (isObject(value)) {
      const s = compressCssSelector(key)
      if (res[s] === undefined) {
        res[s] = {}
      }

      preprocessCssInJs(obj[key], res[s])
    }
  }
  return res
}

export function getSelector(type: string, prefix: string = '-') {
  return type === '' ? type : `${prefix}${type}`
}

export function recursiveNodes(nodes: postcss.ChildNode[], result: Record<string, any> = {}) {
  for (const node of nodes) {
    switch (node.type) {
      case 'atrule': {
        if (node.name === 'apply') {
          const v = get(result, 'apply')
          if (Array.isArray(v)) {
            v.push(node.params)
          } else {
            set(result, 'apply', [node.params])
          }
        } else {
          const selector = `@${node.name} ${node.params}`
          result[selector] = {}
          recursiveNodes(node.nodes, result[selector])
        }

        break
      }
      case 'rule': {
        const s = compressCssSelector(node.selector)
        if (result[s] === undefined) {
          result[s] = {}
        }

        recursiveNodes(node.nodes, result[s])
        break
      }
      case 'decl': {
        set(result, `css.${node.prop}`, node.value + (node.important ? ' !important' : ''))
        break
      }
    }
  }
  return result
}

export function transformCss2Js(css: string) {
  const root = postcss.parse(css)
  const result = recursiveNodes(root.nodes)
  return result
}

export { default as defu } from 'defu'
