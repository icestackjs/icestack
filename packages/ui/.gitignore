assets/css
assets/plugins
assets/js
assets/css-resolved