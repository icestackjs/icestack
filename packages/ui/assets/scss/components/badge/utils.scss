@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("badge.selector");
$DefaultMap: inject("badge.defaults.utils");

#{$S} {
  @each $typeName, $map in map.get($DefaultMap, "sizes") {
    &#{getSelector($typeName)} {
      @include resolve(map.get($map, "default"));
    }
  }
}
