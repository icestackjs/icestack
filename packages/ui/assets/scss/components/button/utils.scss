@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("button.selector");
$DefaultMap: inject("button.defaults.utils");

#{$S} {
  @each $size, $map in map.get($DefaultMap, "sizes") {
    &#{getSelector($size)} {
      @include resolve(map.get($map, "default"));
    }
  }

  @each $shape, $map in map.get($DefaultMap, "shapes") {
    &#{getSelector($shape)} {
      @include resolve(map.get($map, "default"));

      @each $size, $m in map.get($map, "sizes") {
        &:where(#{getSelector($size,#{$S}-)}) {
          @include resolve(map.get($m, "default"));
        }
      }
    }
  }
}
