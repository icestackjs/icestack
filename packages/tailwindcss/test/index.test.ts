import path from 'node:path'
import { createContext } from '@icestack/ui'
import ci from 'ci-info'
import { getCss } from './utils'
import { icestackPlugin } from '@/index'

describe.skipIf(ci.isCI)('tailwindcss', () => {
  const outdir = path.resolve(__dirname, 'assets')
  beforeAll(async () => {
    const ctx = createContext({
      outdir
    })
    await ctx.build()
  })
  it('not throw', () => {
    expect(() => {
      icestackPlugin({})
    }).not.toThrow()
  })

  it('use plugin case 0', async () => {
    const { css } = await getCss({
      content: [
        {
          raw: 'btn'
        }
      ],
      plugins: [
        icestackPlugin({
          loadDirectory: outdir
        })
      ]
    })
    expect(css).toMatchSnapshot()
  })

  // it('use plugin case 1', async () => {
  //   const { css } = await getCss({
  //     content: [
  //       {
  //         raw: 'btn'
  //       }
  //     ],
  //     plugins: [
  //       icestackPlugin({
  //         components: {
  //           button: {
  //             append: [
  //               {
  //                 '.btn::after': {
  //                   border: 'none'
  //                 }
  //               }
  //             ]
  //           }
  //         }
  //       })
  //     ]
  //   })
  //   expect(css).toMatchSnapshot()
  // })

  // it('use plugin case 2', async () => {
  //   const { css } = await getCss({
  //     content: [
  //       {
  //         raw: 'btn'
  //       }
  //     ],
  //     plugins: [
  //       icestackPlugin({
  //         presets: [miniprogramPreset()]
  //       })
  //     ]
  //   })
  //   expect(css).toMatchSnapshot()
  // })

  // const { default: base } = await import('../assets/js/base/index.js')
  // const { default: components } = await import('../assets/js/components/index.js')
  // const { default: utilities } = await import('../assets/js/utilities/index.js')
  // it('grouped components', () => {
  //   const dic = groupBy(Object.entries(components), ([name]) => {
  //     let com = name
  //     if (com.includes('/')) {
  //       com = name.split('/')[1]
  //     }
  //     return com
  //   })
  //   for (const key in dic) {
  //     if (Object.prototype.hasOwnProperty.call(dic, key)) {
  //       const arr = dic[key]
  //       // @ts-ignore
  //       dic[key] = arr.map((x) => x[1])
  //     }
  //   }
  //   expect(dic).toBeTruthy()
  // })

  // it('entries components', () => {
  //   const entries = Object.entries(
  //     groupBy(Object.entries(components), ([name]) => {
  //       let com = name
  //       if (com.includes('/')) {
  //         com = name.split('/')[1]
  //       }
  //       return com
  //     })
  //   ).map(([key, arr]) => {
  //     return [key, arr.map((x) => x[1])]
  //   })
  //   for (const [name, c] of entries) {
  //     if (Array.isArray(c)) {
  //       expect(merge.recursive(true, ...c)).toMatchSnapshot('merged')
  //       for (const cc of c) {
  //         if (typeof cc === 'string') {
  //           console.log(name, cc)
  //         }
  //       }
  //     }
  //   }

  //   expect(entries).toBeTruthy()
  //   // expect(entries).toMatchSnapshot('common')
  // })

  // it('utilitiesEntries', () => {
  //   const utilitiesEntries = Object.entries(
  //     groupBy(Object.entries(utilities), ([name]) => {
  //       let com = name
  //       if (com.includes('/')) {
  //         com = name.split('/')[1]
  //       }
  //       return com
  //     })
  //   ).map(([key, arr]) => {
  //     return [key, arr.map((x) => x[1])]
  //   })

  //   for (const [name, u] of utilitiesEntries) {
  //     if (Array.isArray(u)) {
  //       expect(merge.recursive(true, ...u)).toMatchSnapshot()
  //     }
  //   }
  // })
})
