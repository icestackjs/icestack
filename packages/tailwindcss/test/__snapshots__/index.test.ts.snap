// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tailwindcss > use plugin case 0 1`] = `
":root {--ice-primary: 22 119 255;--ice-primary-hover: 64 150 255;--ice-primary-active: 9 88 217;--ice-primary-content: 255 255 255;--ice-success: 82 196 26;--ice-success-hover: 115 209 61;--ice-success-active: 56 158 13;--ice-success-content: 255 255 255;--ice-warning: 250 173 20;--ice-warning-hover: 255 197 61;--ice-warning-active: 212 136 6;--ice-warning-content: 255 255 255;--ice-error: 245 34 45;--ice-error-hover: 255 77 79;--ice-error-active: 207 19 34;--ice-error-content: 255 255 255;--ice-neutral: 102 102 102;--ice-neutral-hover: 115 115 115;--ice-neutral-active: 64 64 64;--ice-neutral-content: 255 255 255;--ice-secondary: 235 47 150;--ice-secondary-hover: 247 89 171;--ice-secondary-active: 196 29 127;--ice-secondary-content: 255 255 255;--ice-accent: 19 194 194;--ice-accent-hover: 54 207 201;--ice-accent-active: 8 151 156;--ice-accent-content: 255 255 255;--ice-info: 47 84 235;--ice-info-hover: 89 126 247;--ice-info-active: 29 57 196;--ice-info-content: 255 255 255;--ice-base-100: 255 255 255;--ice-base-200: 242 242 242;--ice-base-300: 229 230 230;--ice-base-content: 31 41 55;--ice-ant-neutral-100: 255 255 255;--ice-ant-neutral-200: 250 250 250;--ice-ant-neutral-300: 245 245 245;--ice-ant-neutral-400: 240 240 240;--ice-ant-neutral-500: 217 217 217;--ice-ant-neutral-600: 191 191 191;--ice-ant-neutral-700: 140 140 140;--ice-ant-neutral-800: 89 89 89;--ice-ant-neutral-900: 67 67 67;--ice-ant-neutral-1000: 38 38 38;--ice-ant-neutral-1100: 31 31 31;--ice-ant-neutral-1200: 20 20 20;--ice-ant-neutral-1300: 0 0 0;--ice-rounded-box: 1rem;--ice-rounded-btn: 0.5rem;--ice-rounded-badge: 1.9rem;--ice-animation-btn: 0.25s;--ice-animation-input: 0.2s;--ice-skeleton-duration: 1.2s;--ice-border-btn: 1px;--ice-tab-border: 1px;--ice-tab-radius: 0.5rem 
}[data-mode=dark] {--ice-primary: 22 104 220;--ice-primary-hover: 21 84 173;--ice-primary-active: 60 137 232;--ice-primary-content: 255 255 255;--ice-success: 73 170 25;--ice-success-hover: 60 134 24;--ice-success-active: 106 190 57;--ice-success-content: 255 255 255;--ice-warning: 216 150 20;--ice-warning-hover: 170 119 20;--ice-warning-active: 232 179 57;--ice-warning-content: 255 255 255;--ice-error: 211 32 41;--ice-error-hover: 166 29 36;--ice-error-active: 232 71 73;--ice-error-content: 255 255 255;--ice-neutral: 90 90 90;--ice-neutral-hover: 73 73 73;--ice-neutral-active: 106 106 106;--ice-neutral-content: 255 255 255;--ice-secondary: 203 43 131;--ice-secondary-hover: 160 38 105;--ice-secondary-active: 224 82 156;--ice-secondary-content: 255 255 255;--ice-accent: 19 168 168;--ice-accent-hover: 19 133 133;--ice-accent-active: 51 188 183;--ice-accent-content: 255 255 255;--ice-info: 43 74 203;--ice-info-hover: 38 62 160;--ice-info-active: 82 115 224;--ice-info-content: 255 255 255;--ice-base-100: 29 35 42;--ice-base-200: 25 30 36;--ice-base-300: 21 25 30;--ice-base-content: 166 173 187;--ice-ant-neutral-1300: 255 255 255;--ice-ant-neutral-1200: 250 250 250;--ice-ant-neutral-1100: 245 245 245;--ice-ant-neutral-1000: 240 240 240;--ice-ant-neutral-900: 217 217 217;--ice-ant-neutral-800: 191 191 191;--ice-ant-neutral-700: 140 140 140;--ice-ant-neutral-600: 89 89 89;--ice-ant-neutral-500: 67 67 67;--ice-ant-neutral-400: 38 38 38;--ice-ant-neutral-300: 31 31 31;--ice-ant-neutral-200: 20 20 20;--ice-ant-neutral-100: 0 0 0;--ice-rounded-box: 1rem;--ice-rounded-btn: 0.5rem;--ice-rounded-badge: 1.9rem;--ice-animation-btn: 0.25s;--ice-animation-input: 0.2s;--ice-skeleton-duration: 1.2s;--ice-border-btn: 1px;--ice-tab-border: 1px;--ice-tab-radius: 0.5rem 
}*, ::before, ::after {--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x:  ;--tw-pan-y:  ;--tw-pinch-zoom:  ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position:  ;--tw-gradient-via-position:  ;--tw-gradient-to-position:  ;--tw-ordinal:  ;--tw-slashed-zero:  ;--tw-numeric-figure:  ;--tw-numeric-spacing:  ;--tw-numeric-fraction:  ;--tw-ring-inset:  ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / 0.5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur:  ;--tw-brightness:  ;--tw-contrast:  ;--tw-grayscale:  ;--tw-hue-rotate:  ;--tw-invert:  ;--tw-saturate:  ;--tw-sepia:  ;--tw-drop-shadow:  ;--tw-backdrop-blur:  ;--tw-backdrop-brightness:  ;--tw-backdrop-contrast:  ;--tw-backdrop-grayscale:  ;--tw-backdrop-hue-rotate:  ;--tw-backdrop-invert:  ;--tw-backdrop-opacity:  ;--tw-backdrop-saturate:  ;--tw-backdrop-sepia:  
}::backdrop {--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x:  ;--tw-pan-y:  ;--tw-pinch-zoom:  ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position:  ;--tw-gradient-via-position:  ;--tw-gradient-to-position:  ;--tw-ordinal:  ;--tw-slashed-zero:  ;--tw-numeric-figure:  ;--tw-numeric-spacing:  ;--tw-numeric-fraction:  ;--tw-ring-inset:  ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / 0.5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur:  ;--tw-brightness:  ;--tw-contrast:  ;--tw-grayscale:  ;--tw-hue-rotate:  ;--tw-invert:  ;--tw-saturate:  ;--tw-sepia:  ;--tw-drop-shadow:  ;--tw-backdrop-blur:  ;--tw-backdrop-brightness:  ;--tw-backdrop-contrast:  ;--tw-backdrop-grayscale:  ;--tw-backdrop-hue-rotate:  ;--tw-backdrop-invert:  ;--tw-backdrop-opacity:  ;--tw-backdrop-saturate:  ;--tw-backdrop-sepia:  
}.btn {display: inline-flex;height: 3rem;min-height: 3rem;flex-shrink: 0;cursor: pointer;user-select: none;flex-wrap: wrap;align-items: center;justify-content: center;border-radius: var(--ice-rounded-btn, 0.5rem);border-color: rgba(var(--ice-ant-neutral-400) / var(--tw-border-opacity));padding-left: 1rem;padding-right: 1rem;text-align: center;font-size: 0.875rem;line-height: 1em;--tw-border-opacity: 1;--tw-bg-opacity: 1;background-color: rgba(var(--ice-ant-neutral-400) / var(--tw-bg-opacity));--tw-text-opacity: 1;color: rgba(var(--ice-base-content) / var(--tw-text-opacity));text-decoration-line: none;outline-color: rgba(var(--ice-ant-neutral-400) / 1);border-width: var(--ice-border-btn, 1px)
}.btn-disabled,.btn[disabled],.btn:disabled {pointer-events: none
}:where(.btn:is(input[type=checkbox])),:where(.btn:is(input[type=radio])) {width: auto;appearance: none
}.btn:is(input[type=checkbox]):after,.btn:is(input[type=radio]):after {--tw-content: attr(aria-label);content: var(--tw-content)
}@charset "UTF-8";.btn:focus-visible,.btn.btn-focus-visible {outline-style: solid;outline-width: 2px;outline-offset: 2px
}.btn:hover,.btn.btn-hover {--tw-border-opacity: 1;border-color: rgba(var(--ice-ant-neutral-300) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-ant-neutral-300) / var(--tw-bg-opacity))
}.btn:active,.btn.btn-active {--tw-border-opacity: 1;border-color: rgba(var(--ice-ant-neutral-500) / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color: rgba(var(--ice-ant-neutral-500) / var(--tw-bg-opacity))
}.btn.glass {--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);outline-color: currentColor
}.btn.glass:active,.btn.glass.btn-active {--ice-glass-opacity: 25%； --glass-border-opacity: 15%；
}.btn.btn-disabled,.btn[disabled],.btn:disabled {--tw-border-opacity: 0;background-color: rgba(var(--ice-neutral) / var(--tw-bg-opacity));--tw-bg-opacity: 0.2;color: rgba(var(--ice-base-content) / var(--tw-text-opacity));--tw-text-opacity: 0.2
}@keyframes checkmark {0% {background-position-y: 5px
    }50% {background-position-y: -2px
    }100% {background-position-y: 0
    }
}@keyframes progress-loading {50% {background-position-x: -115%
    }
}@keyframes radiomark {0% {box-shadow: 0 0 0 12px rgba(var(--ice-base-100) / 1) inset, 0 0 0 12px rgba(var(--ice-base-100) / 1) inset
    }50% {box-shadow: 0 0 0 3px rgba(var(--ice-base-100) / 1) inset, 0 0 0 3px rgba(var(--ice-base-100) / 1) inset
    }100% {box-shadow: 0 0 0 4px rgba(var(--ice-base-100) / 1) inset, 0 0 0 4px rgba(var(--ice-base-100) / 1) inset
    }
}@keyframes skeleton-blink {50% {opacity: 0.6
    }
}@keyframes toast-pop {0% {transform: scale(0.9);opacity: 0
    }100% {transform: scale(1);opacity: 1
    }
}@charset "UTF-8";@keyframes rating-pop {0% {transform: translateY(-0.125em)
    }40% {transform: translateY(-0.125em)
    }100% {transform: translateY(0)
    }
}"
`;
